e2e/testcases/stress_test.go

```diff
@@ -261,6 +265,91 @@ func TestStressLargeRequest(t *testing.T) {
 	}
 }
 
+func TestStress100CRDs(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation1, ntopts.Unstructured, ntopts.StressTest,
+		ntopts.WithReconcileTimeout(configsync.DefaultReconcileTimeout))
+
+	// Apply 100 small CRDs on the cluster to make sure Config Sync can still work.
+	for i := 1; i <= 100; i++ {
+		crd := fakeCRD("anvil", fmt.Sprintf("acme-%d.com", i))
+		if err := nt.KubeClient.Create(crd); err != nil {
+			nt.T.Fatal(err)
+		}
+	}
+
+	nt.T.Log("restarting the reconciler Pod to discovery the new 100 CRDs in the controller startup time")
+	nomostest.DeletePodByLabel(nt, "app", reconcilermanager.Reconciler, true)
+
+	nt.T.Log("adding a dummy file to the repo to trigger a new sync")
+	nt.Must(nt.RootRepos[configsync.RootSyncName].AddFile("stress-text.txt", []byte("stress-test")))
```

Comment:
I feel we may as well add a simple k8s resource such as a namespace

Reply:
I guess we don't have to. The new commit will trigger the entire fetch-parse-apply loop even though the resource list is the same.

Reply:
Agreed it shouldn't make much difference, I just figure the test complexity is similar and it ensures that the parse-watch-apply loop has some apply work to do

Reply:
done